Source/hydro/ppm_nd.F90

```diff
@@ -615,108 +633,112 @@ subroutine ppm_type2(s, s_lo, s_hi, ncomp, n, &
     ! compute s at y-edges
 
     ! interpolate s to y-edges
-    do j=ilo2-2, ihi2+3
-       do i=ilo1-1, ihi1+1
-          sedge(i,j) = SEVEN12TH*(s(i,j-1,k3d,n)+s(i,j,k3d,n)) &
-               - TWELFTH*(s(i,j-2,k3d,n)+s(i,j+1,k3d,n))
-          !
-          ! limit sedge
-          !
-          if ((sedge(i,j)-s(i,j-1,k3d,n))*(s(i,j,k3d,n)-sedge(i,j)) .lt. ZERO) then
-             D2  = THREE*(s(i,j-1,k3d,n)-TWO*sedge(i,j)+s(i,j,k3d,n))
-             D2L = s(i,j-2,k3d,n)-TWO*s(i,j-1,k3d,n)+s(i,j,k3d,n)
-             D2R = s(i,j-1,k3d,n)-TWO*s(i,j,k3d,n)+s(i,j+1,k3d,n)
-             sgn = sign(ONE,D2)
-             D2LIM = sgn*max(min(C*sgn*D2L,C*sgn*D2R,sgn*D2),ZERO)
-             sedge(i,j) = HALF*(s(i,j-1,k3d,n)+s(i,j,k3d,n)) - SIXTH*D2LIM
-          end if
+    do k = lo(3)-1, hi(3)+1
```

Comment:
Should this block and the next one have dg(3) guards on the k loop?

Reply:
yes.  Good catch.